raft/log.go

```diff
@@ -1021,9 +1026,11 @@ func (l *Log) candidateLoop(closing <-chan struct{}) State {
 			l.lock()
 			l.mustSetTermIfHigher(hb.term)
 			if hb.term >= l.term {
+				l.tracef("candidateLoop: new leader: old=%d new=%d", l.leaderID, hb.leaderID)
 				l.leaderID = hb.leaderID
 			}
 			l.unlock()
+			return Follower
```

Comment:
So re-reading 5.2:

> While waiting for votes, a candidate may receive an AppendEntries RPC from another server  claiming to be leader. If the leader’s term (included in its RPC) is at least as large as the candidate’s current term, then the candidate recognizes the leader as legitimate and returns to follower state. 

I think you might be correct in that it should only transition if `hb.term >= l.term`.  I'll test that out